src/IonWriter.ts

```diff
@@ -17,43 +17,138 @@ import {Reader} from "./IonReader";
+     * Writes a `boolean` as an Ion `bool` value.
+     *
+     * @param value The `boolean` to write, which may be `null` to write a `null.bool`.
+     */
+    writeBoolean(value: boolean | null) : void;
+
+    /**
+     * Writes a `number` value as an Ion `int`.
+     *
+     * @param value The `number` to write, which may be `null` to write a `null.int`.
+     *  If `number` is not an integer between the range specified by `Number.MIN_SAFE_INTEGER`
+     *  and `Number.MAX_SAFE_INTEGER`, an implementation may truncate or round the value.
+     */
+    writeInt(value: number | null) : void;
+
+    /**
+     * Writes a `number` value as an Ion 32-bit binary `float` value.
+     *
+     * @param value The `number` to write, which may be `null` to write a `null.float`.
+     *  If the `number` cannot be represented exactly as a 32-bit binary floating point
+     *  value, an implementation may truncate or round the value.
+     */
+    writeFloat32(value: number | null) : void;
+
+    /**
+     * Writes a `number` value as an Ion 64-bit binary `float` value.
+     *
+     * @param value The `number` to write, which may be `null` to write a `null.float`.
+     */
+    writeFloat64(value: number | null) : void;
+
+    /**
+     * Writes a [[Decimal]] value as an Ion `decimal` value.
+     *
+     * @param value The [[Decimal]] to write, which may be `null` to write a `null.decimal`.
+     */
+    writeDecimal(value: Decimal | null) : void;
+
+    writeTimestamp(value: Timestamp) : void;
+
+    /**
+     * Writes a `string` value as an Ion `string`.
+     *
+     * @param value The `string` to write, which may be `null` to write a `null.string`.
+     */
+    writeString(value: string | null) : void;
+
+    /**
+     * Writes a `string` value as an Ion `symbol`.
+     *
+     * @param value The `string` to write, which may be `null` to write a `null.symbol`.
+     */
+    writeSymbol(value: string | null) : void;
+
+    /**
+     * Writes a `Uint8Array` as an Ion `blob` value.
+     *
+     * @param value The array to write, which may be `null` to write a `null.blob`.
+     */
+    writeBlob(value: Uint8Array | null) : void;
+
+    /**
+     * Writes a `Uint8Array` as an Ion `clob` value.
+     *
+     * @param value The array to write, which may be `null` to write a `null.clob`.
+     */
+    writeClob(value: Uint8Array | null) : void;
+
+    /**
+     * Writes a reader's current value and all following values until the end
+     * of the current container.  If there's no current value then this method
+     * calls {@link next()} to get started.
+     */
+    writeValues(reader: Reader) : void;
+
+    /**
+     * Writes the field name for a member of a `struct`.
+     *
+     * @throws Error if the [[Writer]] is not within a `struct` or this method was already
+     *  called before a value was written.
+     */
+    writeFieldName(fieldName: string) : void;
+
+    /**
+     * Starts a container and positions the writer within that container.
```

Comment:
For `stepIn` and `stepOut`, it might be nice to mention that you still have to call `next()` before you can read the following value at whichever depth you now find yourself. We could also address this by having good examples in the `README.md`. Or both!

Reply:
Whoops, sure did. Sorry about that.